c-dependencies/js-compute-runtime/xqd-world/xqd_world_adapter.cc

```diff
@@ -0,0 +1,926 @@
+#include "xqd_world_adapter.h"
+
+JSContext *context = nullptr;
+
+void set_js_context(JSContext *cx) { context = cx; }
+
+#ifndef COMPONENT
```

Comment:
The non-component build using the component-generated APIs is handled by the wrapper functions for `xqd.h` here.